library/src/animationManager.cxx

```diff
@@ -184,24 +201,62 @@ void animationManager::ToggleAnimation()
 //----------------------------------------------------------------------------
 void animationManager::Tick()
 {
-  if (this->HasAnimation)
+  assert(this->Interactor);
+
+  // Compute time since previous tick
+  std::chrono::steady_clock::time_point tick = std::chrono::steady_clock::now();
+  auto timeInMS =
+    std::chrono::duration_cast<std::chrono::milliseconds>(tick - this->PreviousTick).count();
+  this->PreviousTick = tick;
+
+  // Convert to a usable time in seconds
+  double elapsedTime = static_cast<double>(timeInMS) / 1000.0;
+  double animationSpeedFactor = this->Options->getAsDouble("scene.animation.speed-factor");
+  elapsedTime *= animationSpeedFactor;
+
+  // elapsedTime can be negative
+  this->CurrentTime += elapsedTime;
+  if (this->CurrentTime < this->TimeRange[0])
+  {
+    this->CurrentTime = this->TimeRange[1];
+  }
+  else if (this->CurrentTime > this->TimeRange[1])
+  {
+    this->CurrentTime = this->TimeRange[0];
+  }
```

Comment:
Shouldn't the delta past the bounds be added when wrapping around?
If so can the 2 `if` be replaced by a modulo operation?